topology/mask.h

```diff
@@ -733,42 +745,68 @@ EllipseMask< D >::EllipseMask( const DictionaryDatum& d )
       "major_axis greater than minor_axis required." );
   }
 
-  x_dividend_ = 1.0 / ( major_axis_ * major_axis_ );
-  y_dividend_ = 1.0 / ( minor_axis_ * minor_axis_ );
+  x_scale_ = 4.0 / ( major_axis_ * major_axis_ );
+  y_scale_ = 4.0 / ( minor_axis_ * minor_axis_ );
 
-  if ( d->known( names::intermediate_axis ) )
+  if ( d->known( names::polar_axis ) )
   {
-    intermediate_axis_ = getValue< double >( d, names::intermediate_axis );
+    if ( D == 2 )
+    {
+      throw BadProperty(
+        "topology::EllipseMask<D>: "
+        "polar_axis not defined in 2D." );
+    }
+    polar_axis_ = getValue< double >( d, names::polar_axis );
 
-    if ( intermediate_axis_ < 0 )
+    if ( polar_axis_ <= 0 )
     {
       throw BadProperty(
         "topology::EllipseMask<D>: "
         "All axis > 0 required." );
     }
 
-    z_dividend_ = 1.0 / ( intermediate_axis_ * intermediate_axis_ );
+    z_scale_ = 4.0 / ( polar_axis_ * polar_axis_ );
   }
   else
   {
-    z_dividend_ = 0.0;
+    z_scale_ = 0.0;
```

Comment:
Should't `polar_axis_` be set to some dummy value, to? I think 0 makes most sense.